In [1]:
import pandas as pd
df = pd.read_csv("asthma_disease_data.csv")

### Bizda 29ta ustun bor. hammasi ko'rinmayda. Qanday qilib barchasini ko'rsatish mumkin?

In [2]:
pd.set_option("display.max_columns", None)

In [3]:
df.head()

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0       5034   63       0          1               0  15.848744        0   
1       5035   26       1          2               2  22.757042        0   
2       5036   57       0          2               1  18.395396        0   
3       5037   40       1          2               1  38.515278        0   
4       5038   61       0          0               3  19.283802        0   

   PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
0          0.894448     5.488696      8.701003           7.388481   
1          5.897329     6.341014      5.153966           1.969838   
2          6.739367     9.196237      6.840647           1.460593   
3          1.404503     5.826532      4.253036           0.581905   
4          4.604493     3.127048      9.625799           0.980875   

   PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
0        2.855578      0.974339           1                    1   
1        7.457665      6.584631           0                    0   
2        1.448189      5.445799           0                    1   
3        7.571845      3.965316           0                    0   
4        3.049807      8.260605           0                    0   

   HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
0                   0       0         0                       0   
1                   1       0         0                       0   
2                   1       0         1                       0   
3                   0       0         1                       0   
4                   0       0         1                       0   

   LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
0          1.369051         4.941206         0                  0   
1          2.197767         1.702393         1                  0   
2          1.698011         5.022553         1                  1   
3          3.032037         2.300159         1                  0   
4          3.470589         3.067944         1                  1   

   ChestTightness  Coughing  NighttimeSymptoms  ExerciseInduced  Diagnosis  \
0               1         0                  0                1          0   
1               0         1                  1                1          0   
2               1         0                  1                1          0   
3               1         1                  1                0          0   
4               1         0                  0                1          0   

  DoctorInCharge  
0      Dr_Confid  
1      Dr_Confid  
2      Dr_Confid  
3      Dr_Confid  
4      Dr_Confid

ko'rib turganimizday barcha ustunlarimiz ko'rindi. Qulaylik uchun head funksiyasidan ham foydalandik

### Barcha ustunlarni o'zini ko'rishni yana bir usuli

In [4]:
df.columns

Index(['PatientID', 'Age', 'Gender', 'Ethnicity', 'EducationLevel', 'BMI',
       'Smoking', 'PhysicalActivity', 'DietQuality', 'SleepQuality',
       'PollutionExposure', 'PollenExposure', 'DustExposure', 'PetAllergy',
       'FamilyHistoryAsthma', 'HistoryOfAllergies', 'Eczema', 'HayFever',
       'GastroesophagealReflux', 'LungFunctionFEV1', 'LungFunctionFVC',
       'Wheezing', 'ShortnessOfBreath', 'ChestTightness', 'Coughing',
       'NighttimeSymptoms', 'ExerciseInduced', 'Diagnosis', 'DoctorInCharge'],
      dtype='object')

ustunlarning nomlari: 
bemorning ID si, 
yoshi, 
jinsi, 
etnik kelib chiqishi, 
ta'lim darajasi, 
BMI, chekadimi, 
jismoniy faolligi, 
dietasinig sifati, 
uyqu sifati, 
ifloslanish ta'siri, 
Polen ta'sir qilishi(allegiya turi), 
chang ta'sir qilishi, 
uy hayvonlariga allergiyasi, 
oila tarixida astma, 
allergiya tarixi, 
ekzema(ter kasalligi turi), 
pichan isitmasi(allergiya turi), 
gastroezofegal reflyuks(oshqozondagi ovqatni doimo og'ziga kelishi xronik kasallik), 
o'pka funktsiyasi fev 1(1 sekundda qancha hajm havoni o'pkadan chiqishi), 
o'pka funktsiyasi fvc(kuch bilan qancha hajm havoni o'pkadan chiqishi), 
xirillash, 
nafas qisilishi, 
ko'krak qafasi qisilishi,
yo'tal, 
tungi vaqt belgilari, 
jismoniy mashqlar, 
tashxis, 
mas'ul shifokor

In [5]:
df.T

0          1          2          3          4     \
PatientID                    5034       5035       5036       5037       5038   
Age                            63         26         57         40         61   
Gender                          0          1          0          1          0   
Ethnicity                       1          2          2          2          0   
EducationLevel                  0          2          1          1          3   
BMI                     15.848744  22.757042  18.395396  38.515278  19.283802   
Smoking                         0          0          0          0          0   
PhysicalActivity         0.894448   5.897329   6.739367   1.404503   4.604493   
DietQuality              5.488696   6.341014   9.196237   5.826532   3.127048   
SleepQuality             8.701003   5.153966   6.840647   4.253036   9.625799   
PollutionExposure        7.388481   1.969838   1.460593   0.581905   0.980875   
PollenExposure           2.855578   7.457665   1.448189   7.571845   3.049807   
DustExposure             0.974339   6.584631   5.445799   3.965316   8.260605   
PetAllergy                      1          0          0          0          0   
FamilyHistoryAsthma             1          0          1          0          0   
HistoryOfAllergies              0          1          1          0          0   
Eczema                          0          0          0          0          0   
HayFever                        0          0          1          1          1   
GastroesophagealReflux          0          0          0          0          0   
LungFunctionFEV1         1.369051   2.197767   1.698011   3.032037   3.470589   
LungFunctionFVC          4.941206   1.702393   5.022553   2.300159   3.067944   
Wheezing                        0          1          1          1          1   
ShortnessOfBreath               0          0          1          0          1   
ChestTightness                  1          0          1          1          1   
Coughing                        0          1          0          1          0   
NighttimeSymptoms               0          1          1          1          0   
ExerciseInduced                 1          1          1          0          1   
Diagnosis                       0          0          0          0          0   
DoctorInCharge          Dr_Confid  Dr_Confid  Dr_Confid  Dr_Confid  Dr_Confid   

                             5          6          7          8          9     \
PatientID                    5039       5040       5041       5042       5043   
Age                            21         45         26         49         45   
Gender                          0          1          0          1          1   
Ethnicity                       2          1          0          1          1   
EducationLevel                  0          1          1          2          1   
BMI                     21.812975  30.245954  26.048416  32.676204  29.910298   
Smoking                         0          1          1          0          0   
PhysicalActivity         0.470044   9.371784   8.344096   2.690256    2.89572   
DietQuality              1.759118   7.030507   1.626484   3.920034     2.6077   
SleepQuality             9.549262   5.746128   6.431179   5.843645   7.234908   
PollutionExposure        1.711446   7.664306   6.939046   3.180421   1.711722   
PollenExposure           7.192424     5.0465   8.417881   7.292903   0.741449   
DustExposure             6.832048   8.918884   2.190541   8.568676    6.46679   
PetAllergy                      1          0          0          1          0   
FamilyHistoryAsthma             0          0          0          0          0   
HistoryOfAllergies              0          0          0          0          0   
Eczema                          1          0          1          0          0   
HayFever                        0          0          0          0          0   
GastroesophagealReflux          0          0          0    

### Pivot table da ishlash. 

In [6]:
df.groupby("Age")["BMI"].mean()

Age
5     27.466987
6     25.796679
7     27.434039
8     27.841121
9     26.380403
        ...    
75    28.240111
76    25.371413
77    26.806620
78    26.838883
79    28.808912
Name: BMI, Length: 75, dtype: float64

In [7]:
df.pivot_table(values = "BMI", index = "Age")

BMI
Age           
5    27.466987
6    25.796679
7    27.434039
8    27.841121
9    26.380403
..         ...
75   28.240111
76   25.371413
77   26.806620
78   26.838883
79   28.808912

[75 rows x 1 columns]

In [8]:
import numpy as np
df.pivot_table(values = "BMI", index = "Age", aggfunc=np.median)

C:\Users\Rudy\AppData\Local\Temp\ipykernel_17420\2180639155.py:2: FutureWarning: The provided callable <function median at 0x000001DD89779260> is currently using DataFrameGroupBy.median. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "median" instead.
  df.pivot_table(values = "BMI", index = "Age", aggfunc=np.median)


BMI
Age           
5    26.804660
6    24.329249
7    27.553803
8    27.216382
9    26.394701
..         ...
75   29.095900
76   23.727851
77   28.398475
78   27.869366
79   29.993441

[75 rows x 1 columns]

In [9]:
df.pivot_table(values = "BMI", index = "Age", aggfunc = [np.median, np.mean])

C:\Users\Rudy\AppData\Local\Temp\ipykernel_17420\826996512.py:1: FutureWarning: The provided callable <function median at 0x000001DD89779260> is currently using DataFrameGroupBy.median. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "median" instead.
  df.pivot_table(values = "BMI", index = "Age", aggfunc = [np.median, np.mean])
C:\Users\Rudy\AppData\Local\Temp\ipykernel_17420\826996512.py:1: FutureWarning: The provided callable <function mean at 0x000001DD89620F40> is currently using DataFrameGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "mean" instead.
  df.pivot_table(values = "BMI", index = "Age", aggfunc = [np.median, np.mean])


median       mean
           BMI        BMI
Age                      
5    26.804660  27.466987
6    24.329249  25.796679
7    27.553803  27.434039
8    27.216382  27.841121
9    26.394701  26.380403
..         ...        ...
75   29.095900  28.240111
76   23.727851  25.371413
77   28.398475  26.806620
78   27.869366  26.838883
79   29.993441  28.808912

[75 rows x 2 columns]

In [10]:
df.pivot_table(values = "BMI", index = "EducationLevel", columns = "Age")

Age                    5          6          7          8          9   \
EducationLevel                                                          
0               30.094981  24.363593  28.154508  26.413924  22.360115   
1               25.053760  26.005778  26.258252  27.094093  25.427555   
2               29.268822  24.432536  28.077906  26.637354  28.559363   
3               27.276474  31.056175  28.309732  37.462874  32.107553   

Age                    10         11         12         13         14  \
EducationLevel                                                          
0               29.523057  28.565018  27.206367  26.594891  29.605628   
1               28.480458  26.863214  25.975980  30.388063  29.435809   
2               26.745523  30.048083  27.602660  24.414959  24.762988   
3               28.776078  26.510193        NaN  25.887700  15.490855   

Age                    15         16         17         18         19  \
EducationLevel                                                          
0               23.288650  25.146252  28.238133  32.705099  26.624554   
1               26.118586  25.932745  25.466144  26.955002  26.834206   
2               25.684002  23.092863  28.667498  25.390098  28.726721   
3               24.681585  19.329257  23.312611  38.970364  32.424420   

Age                    20         21         22         23         24  \
EducationLevel                                                          
0               28.680585  31.016535  31.718103  21.747253  29.246398   
1               25.157303  29.744347  26.930208  26.328840  27.774472   
2               29.753743  26.205383  25.196742  25.992152  29.426621   
3               29.282822  27.307337  29.273455  28.372235  16.994534   

Age                    25         26         27         28         29  \
EducationLevel                                                          
0               23.642459  28.303200  30.919053  23.726513  26.879494   
1               26.336362  27.485867  26.898890  27.323288  27.071031   
2               25.060498  23.519411  30.208155  28.613343  26.079254   
3               33.774561  28.380926  23.549951  30.633028        NaN   

Age                    30         31         32         33         34  \
EducationLevel                                                          
0               26.027606  25.605373  23.777448  28.058966  26.717475   
1               26.406871  28.160331  24.912164  20.936719  21.255684   
2               24.121519  27.221120  26.728200  28.884701  28.968237   
3               23.777711  36.814369  20.610083  30.595137  28.573950   

Age                    35         36         37         38         39  \
EducationLevel                                                          
0               28.917233  26.595385  30.200496  25.290027  27.280808   
1               27.279658  22.455816  33.144746  23.625553  27.110999   
2               32.887432  25.301687  28.357605  26.640788  28.950856   
3               27.772637  22.436573  31.699431  28.357935  29.024315   

Age                    40         41         42         43         44  \
EducationLevel                                                          
0               31.728635  28.822009  27.539371  29.212104  31.338848   
1               29.564346  23.367638  26.293236  26.338145  26.253171   
2               26.206711  29.719414  31.193825  26.973454  27.649783   
3               25.113713  28.778958  22.031103  25.456598  31.468623   

Age                    45         46         47         48         49  \
EducationLevel                                                          
0               25.490641  29.405044  26.897072  20.618951  31.117762   
1               26.328817  30.191352  27.792434  28.052853  27.013491   
2               30.463215  28.692445  30.539825  20.349506  24.222226   
3               22.467473  24.107703  27.674112  25.412323  17.805883   

Age                    50         51         52  

In [11]:
df.pivot_table(values = "BMI", index = "EducationLevel", columns = "Age", fill_value= 0)

Age                    5          6          7          8          9   \
EducationLevel                                                          
0               30.094981  24.363593  28.154508  26.413924  22.360115   
1               25.053760  26.005778  26.258252  27.094093  25.427555   
2               29.268822  24.432536  28.077906  26.637354  28.559363   
3               27.276474  31.056175  28.309732  37.462874  32.107553   

Age                    10         11         12         13         14  \
EducationLevel                                                          
0               29.523057  28.565018  27.206367  26.594891  29.605628   
1               28.480458  26.863214  25.975980  30.388063  29.435809   
2               26.745523  30.048083  27.602660  24.414959  24.762988   
3               28.776078  26.510193   0.000000  25.887700  15.490855   

Age                    15         16         17         18         19  \
EducationLevel                                                          
0               23.288650  25.146252  28.238133  32.705099  26.624554   
1               26.118586  25.932745  25.466144  26.955002  26.834206   
2               25.684002  23.092863  28.667498  25.390098  28.726721   
3               24.681585  19.329257  23.312611  38.970364  32.424420   

Age                    20         21         22         23         24  \
EducationLevel                                                          
0               28.680585  31.016535  31.718103  21.747253  29.246398   
1               25.157303  29.744347  26.930208  26.328840  27.774472   
2               29.753743  26.205383  25.196742  25.992152  29.426621   
3               29.282822  27.307337  29.273455  28.372235  16.994534   

Age                    25         26         27         28         29  \
EducationLevel                                                          
0               23.642459  28.303200  30.919053  23.726513  26.879494   
1               26.336362  27.485867  26.898890  27.323288  27.071031   
2               25.060498  23.519411  30.208155  28.613343  26.079254   
3               33.774561  28.380926  23.549951  30.633028   0.000000   

Age                    30         31         32         33         34  \
EducationLevel                                                          
0               26.027606  25.605373  23.777448  28.058966  26.717475   
1               26.406871  28.160331  24.912164  20.936719  21.255684   
2               24.121519  27.221120  26.728200  28.884701  28.968237   
3               23.777711  36.814369  20.610083  30.595137  28.573950   

Age                    35         36         37         38         39  \
EducationLevel                                                          
0               28.917233  26.595385  30.200496  25.290027  27.280808   
1               27.279658  22.455816  33.144746  23.625553  27.110999   
2               32.887432  25.301687  28.357605  26.640788  28.950856   
3               27.772637  22.436573  31.699431  28.357935  29.024315   

Age                    40         41         42         43         44  \
EducationLevel                                                          
0               31.728635  28.822009  27.539371  29.212104  31.338848   
1               29.564346  23.367638  26.293236  26.338145  26.253171   
2               26.206711  29.719414  31.193825  26.973454  27.649783   
3               25.113713  28.778958  22.031103  25.456598  31.468623   

Age                    45         46         47         48         49  \
EducationLevel                                                          
0               25.490641  29.405044  26.897072  20.618951  31.117762   
1               26.328817  30.191352  27.792434  28.052853  27.013491   
2               30.463215  28.692445  30.539825  20.349506  24.222226   
3               22.467473  24.107703  27.674112  25.412323  17.805883   

Age                    50         51         52  

fill_value= 0 -> NaN degan kataklarni 0 bilan to'ldiradi

In [12]:
df.pivot_table(values = "BMI", index = "EducationLevel", columns = "Age", fill_value= 0, margins = True)


Age                     5          6          7          8          9  \
EducationLevel                                                          
0               30.094981  24.363593  28.154508  26.413924  22.360115   
1               25.053760  26.005778  26.258252  27.094093  25.427555   
2               29.268822  24.432536  28.077906  26.637354  28.559363   
3               27.276474  31.056175  28.309732  37.462874  32.107553   
All             27.466987  25.796679  27.434039  27.841121  26.380403   

Age                    10         11         12         13         14  \
EducationLevel                                                          
0               29.523057  28.565018  27.206367  26.594891  29.605628   
1               28.480458  26.863214  25.975980  30.388063  29.435809   
2               26.745523  30.048083  27.602660  24.414959  24.762988   
3               28.776078  26.510193   0.000000  25.887700  15.490855   
All             28.242241  28.315275  26.936220  27.832035  27.229759   

Age                    15         16         17         18         19  \
EducationLevel                                                          
0               23.288650  25.146252  28.238133  32.705099  26.624554   
1               26.118586  25.932745  25.466144  26.955002  26.834206   
2               25.684002  23.092863  28.667498  25.390098  28.726721   
3               24.681585  19.329257  23.312611  38.970364  32.424420   
All             25.377354  24.095951  27.134357  28.327557  28.068706   

Age                    20         21         22         23         24  \
EducationLevel                                                          
0               28.680585  31.016535  31.718103  21.747253  29.246398   
1               25.157303  29.744347  26.930208  26.328840  27.774472   
2               29.753743  26.205383  25.196742  25.992152  29.426621   
3               29.282822  27.307337  29.273455  28.372235  16.994534   
All             27.939231  28.322474  27.260759  26.237603  28.235513   

Age                    25         26         27         28         29  \
EducationLevel                                                          
0               23.642459  28.303200  30.919053  23.726513  26.879494   
1               26.336362  27.485867  26.898890  27.323288  27.071031   
2               25.060498  23.519411  30.208155  28.613343  26.079254   
3               33.774561  28.380926  23.549951  30.633028   0.000000   
All             26.919173  26.425813  28.644535  27.620328  26.688209   

Age                    30         31         32         33         34  \
EducationLevel                                                          
0               26.027606  25.605373  23.777448  28.058966  26.717475   
1               26.406871  28.160331  24.912164  20.936719  21.255684   
2               24.121519  27.221120  26.728200  28.884701  28.968237   
3               23.777711  36.814369  20.610083  30.595137  28.573950   
All             25.315843  28.200340  25.118308  25.326050  26.560749   

Age                    35         36         37         38         39  \
EducationLevel                                                          
0               28.917233  26.595385  30.200496  25.290027  27.280808   
1               27.279658  22.455816  33.144746  23.625553  27.110999   
2               32.887432  25.301687  28.357605  26.640788  28.950856   
3               27.772637  22.436573  31.699431  28.357935  29.024315   
All             28.913064  23.989579  29.844968  25.305508  28.303511   

Age                    40         41         42         43         44  \
EducationLevel                                                          
0               31.728635  28.822009  27.539371  29.212104  31.338848   
1               29.564346  23.367638  26.293236  26.338145  26.253171   
2               26.206711  29.719414  31.193825  26.973454  27.649783   
3               25.113713  28.778958  22.031103  25

### Index ni ko'rish

In [13]:
df.index

RangeIndex(start=0, stop=2392, step=1)

Index 0 dan boshlanib 2392 da to'xtayabti. 1tadan yuribdi

### Ustunni index qilib olish yo'llarini ko'ramiz. Bunda olayotgan ustunimizning obyektlari qaytarilmasligiga ishonch hosil qilmog'imiz zarur

#### PatientID

In [14]:
df.shape

(2392, 29)

In [15]:
df_ID = df.set_index("PatientID")
df_ID.head()

Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
PatientID                                                               
5034        63       0          1               0  15.848744        0   
5035        26       1          2               2  22.757042        0   
5036        57       0          2               1  18.395396        0   
5037        40       1          2               1  38.515278        0   
5038        61       0          0               3  19.283802        0   

           PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
PatientID                                                                   
5034               0.894448     5.488696      8.701003           7.388481   
5035               5.897329     6.341014      5.153966           1.969838   
5036               6.739367     9.196237      6.840647           1.460593   
5037               1.404503     5.826532      4.253036           0.581905   
5038               4.604493     3.127048      9.625799           0.980875   

           PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
PatientID                                                                  
5034             2.855578      0.974339           1                    1   
5035             7.457665      6.584631           0                    0   
5036             1.448189      5.445799           0                    1   
5037             7.571845      3.965316           0                    0   
5038             3.049807      8.260605           0                    0   

           HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
PatientID                                                                 
5034                        0       0         0                       0   
5035                        1       0         0                       0   
5036                        1       0         1                       0   
5037                        0       0         1                       0   
5038                        0       0         1                       0   

           LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
PatientID                                                                   
5034               1.369051         4.941206         0                  0   
5035               2.197767         1.702393         1                  0   
5036               1.698011         5.022553         1                  1   
5037               3.032037         2.300159         1                  0   
5038               3.470589         3.067944         1                  1   

           ChestTightness  Coughing  NighttimeSymptoms  ExerciseInduced  \
PatientID                                                                 
5034                    1         0                  0                1   
5035                    0         1                  1                1   
5036                    1         0                  1                1   
5037                    1         1                  1                0   
5038                    1         0                  0                1   

           Diagnosis DoctorInCharge  
PatientID                            
5034               0      Dr_Confid  
5035               0      Dr_Confid  
5036               0      Dr_Confid  
5037               0      Dr_Confid  
5038               0      Dr_Confid

In [16]:
df_ID.shape

(2392, 28)

In [17]:
df.drop_duplicates("PatientID").shape

(2392, 29)

#### BMI

In [18]:
df_BMI = df.set_index("BMI")
df_BMI.head()

PatientID  Age  Gender  Ethnicity  EducationLevel  Smoking  \
BMI                                                                     
15.848744       5034   63       0          1               0        0   
22.757042       5035   26       1          2               2        0   
18.395396       5036   57       0          2               1        0   
38.515278       5037   40       1          2               1        0   
19.283802       5038   61       0          0               3        0   

           PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
BMI                                                                         
15.848744          0.894448     5.488696      8.701003           7.388481   
22.757042          5.897329     6.341014      5.153966           1.969838   
18.395396          6.739367     9.196237      6.840647           1.460593   
38.515278          1.404503     5.826532      4.253036           0.581905   
19.283802          4.604493     3.127048      9.625799           0.980875   

           PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
BMI                                                                        
15.848744        2.855578      0.974339           1                    1   
22.757042        7.457665      6.584631           0                    0   
18.395396        1.448189      5.445799           0                    1   
38.515278        7.571845      3.965316           0                    0   
19.283802        3.049807      8.260605           0                    0   

           HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
BMI                                                                       
15.848744                   0       0         0                       0   
22.757042                   1       0         0                       0   
18.395396                   1       0         1                       0   
38.515278                   0       0         1                       0   
19.283802                   0       0         1                       0   

           LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
BMI                                                                         
15.848744          1.369051         4.941206         0                  0   
22.757042          2.197767         1.702393         1                  0   
18.395396          1.698011         5.022553         1                  1   
38.515278          3.032037         2.300159         1                  0   
19.283802          3.470589         3.067944         1                  1   

           ChestTightness  Coughing  NighttimeSymptoms  ExerciseInduced  \
BMI                                                                       
15.848744               1         0                  0                1   
22.757042               0         1                  1                1   
18.395396               1         0                  1                1   
38.515278               1         1                  1                0   
19.283802               1         0                  0                1   

           Diagnosis DoctorInCharge  
BMI                                  
15.848744          0      Dr_Confid  
22.757042          0      Dr_Confid  
18.395396          0      Dr_Confid  
38.515278          0      Dr_Confid  
19.283802          0      Dr_Confid

In [19]:
df_BMI.shape

(2392, 28)

In [20]:
df.drop_duplicates("BMI").shape

(2392, 29)

#### Age

In [21]:
df_Age = df.set_index("Age")
df_Age.head()

PatientID  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
Age                                                                     
63        5034       0          1               0  15.848744        0   
26        5035       1          2               2  22.757042        0   
57        5036       0          2               1  18.395396        0   
40        5037       1          2               1  38.515278        0   
61        5038       0          0               3  19.283802        0   

     PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
Age                                                                   
63           0.894448     5.488696      8.701003           7.388481   
26           5.897329     6.341014      5.153966           1.969838   
57           6.739367     9.196237      6.840647           1.460593   
40           1.404503     5.826532      4.253036           0.581905   
61           4.604493     3.127048      9.625799           0.980875   

     PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
Age                                                                  
63         2.855578      0.974339           1                    1   
26         7.457665      6.584631           0                    0   
57         1.448189      5.445799           0                    1   
40         7.571845      3.965316           0                    0   
61         3.049807      8.260605           0                    0   

     HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
Age                                                                 
63                    0       0         0                       0   
26                    1       0         0                       0   
57                    1       0         1                       0   
40                    0       0         1                       0   
61                    0       0         1                       0   

     LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
Age                                                                   
63           1.369051         4.941206         0                  0   
26           2.197767         1.702393         1                  0   
57           1.698011         5.022553         1                  1   
40           3.032037         2.300159         1                  0   
61           3.470589         3.067944         1                  1   

     ChestTightness  Coughing  NighttimeSymptoms  ExerciseInduced  Diagnosis  \
Age                                                                            
63                1         0                  0                1          0   
26                0         1                  1                1          0   
57                1         0                  1                1          0   
40                1         1                  1                0          0   
61                1         0                  0                1          0   

    DoctorInCharge  
Age                 
63       Dr_Confid  
26       Dr_Confid  
57       Dr_Confid  
40       Dr_Confid  
61       Dr_Confid

In [22]:
df_Age.shape

(2392, 28)

In [23]:
df.drop_duplicates("Age").shape

(75, 29)

Ko'rib turganimizdey ustunni muvaffaqiyatli indexga aylantirdik. Ammo "PatientID" va "BMI" dublikatlar bo'lmasada, "Age" ya'ni "Yosh" da dublikatlar bor va bu keyinchalik xatoga olib kelishi mumkin

### 0 dan boshlanadigan standart Indexni qaytarsa bo'ladimi?

In [24]:
df_ID.reset_index()

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0          5034   63       0          1               0  15.848744        0   
1          5035   26       1          2               2  22.757042        0   
2          5036   57       0          2               1  18.395396        0   
3          5037   40       1          2               1  38.515278        0   
4          5038   61       0          0               3  19.283802        0   
...         ...  ...     ...        ...             ...        ...      ...   
2387       7421   43       1          0               2  29.059613        0   
2388       7422   18       1          0               1  20.740850        0   
2389       7423   54       0          3               2  37.079560        0   
2390       7424   46       1          0               2  23.444712        0   
2391       7425   26       1          0               0  28.123021        1   

      PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
0             0.894448     5.488696      8.701003           7.388481   
1             5.897329     6.341014      5.153966           1.969838   
2             6.739367     9.196237      6.840647           1.460593   
3             1.404503     5.826532      4.253036           0.581905   
4             4.604493     3.127048      9.625799           0.980875   
...                ...          ...           ...                ...   
2387          3.019854     6.119637      8.300960           2.483829   
2388          5.805180     4.386992      7.731192           7.733983   
2389          4.735169     8.214064      7.483521           2.794847   
2390          9.672637     7.362861      6.717272           9.448862   
2391          1.613138     7.412878      8.512253           3.231709   

      PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
0           2.855578      0.974339           1                    1   
1           7.457665      6.584631           0                    0   
2           1.448189      5.445799           0                    1   
3           7.571845      3.965316           0                    0   
4           3.049807      8.260605           0                    0   
...              ...           ...         ...                  ...   
2387        7.314582      3.425445           1                    0   
2388        2.279073      6.467701           0                    0   
2389        3.055139      9.484013           0                    0   
2390        7.712584      5.051405           0                    1   
2391        3.874028      5.064317           0                    1   

      HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
0                      0       0         0                       0   
1                      1       0         0                       0   
2                      1       0         1                       0   
3                      0       0         1                       0   
4                      0       0         1                       0   
...                  ...     ...       ...                     ...   
2387                   1       0         0                       0   
2388                   0       1         0                       0   
2389                   0       0         1                       0   
2390                   1       0         0                       1   
2391                   0       0         0                       0   

      LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
0             1.369051         4.941206         0                  0   
1             2.197767         1.702393         1                  0   
2             1.698011         5.022553         1                  1   
3             3.032037         2.300159         1                  0   
4             3.470589         3.067944         1                  1   
...                ...              ...       ...                .

In [25]:
df_BMI.reset_index().head()

BMI  PatientID  Age  Gender  Ethnicity  EducationLevel  Smoking  \
0  15.848744       5034   63       0          1               0        0   
1  22.757042       5035   26       1          2               2        0   
2  18.395396       5036   57       0          2               1        0   
3  38.515278       5037   40       1          2               1        0   
4  19.283802       5038   61       0          0               3        0   

   PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
0          0.894448     5.488696      8.701003           7.388481   
1          5.897329     6.341014      5.153966           1.969838   
2          6.739367     9.196237      6.840647           1.460593   
3          1.404503     5.826532      4.253036           0.581905   
4          4.604493     3.127048      9.625799           0.980875   

   PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
0        2.855578      0.974339           1                    1   
1        7.457665      6.584631           0                    0   
2        1.448189      5.445799           0                    1   
3        7.571845      3.965316           0                    0   
4        3.049807      8.260605           0                    0   

   HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
0                   0       0         0                       0   
1                   1       0         0                       0   
2                   1       0         1                       0   
3                   0       0         1                       0   
4                   0       0         1                       0   

   LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
0          1.369051         4.941206         0                  0   
1          2.197767         1.702393         1                  0   
2          1.698011         5.022553         1                  1   
3          3.032037         2.300159         1                  0   
4          3.470589         3.067944         1                  1   

   ChestTightness  Coughing  NighttimeSymptoms  ExerciseInduced  Diagnosis  \
0               1         0                  0                1          0   
1               0         1                  1                1          0   
2               1         0                  1                1          0   
3               1         1                  1                0          0   
4               1         0                  0                1          0   

  DoctorInCharge  
0      Dr_Confid  
1      Dr_Confid  
2      Dr_Confid  
3      Dr_Confid  
4      Dr_Confid

Ko'rib turganimizdek bo'lar ekan. PayientID va BMI ham tashlab yuborilmadi. Endi Olingan indexni tashlab yuborishni ko'ramiz.

In [26]:
df_Age.reset_index(drop=True).head()

PatientID  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0       5034       0          1               0  15.848744        0   
1       5035       1          2               2  22.757042        0   
2       5036       0          2               1  18.395396        0   
3       5037       1          2               1  38.515278        0   
4       5038       0          0               3  19.283802        0   

   PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
0          0.894448     5.488696      8.701003           7.388481   
1          5.897329     6.341014      5.153966           1.969838   
2          6.739367     9.196237      6.840647           1.460593   
3          1.404503     5.826532      4.253036           0.581905   
4          4.604493     3.127048      9.625799           0.980875   

   PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
0        2.855578      0.974339           1                    1   
1        7.457665      6.584631           0                    0   
2        1.448189      5.445799           0                    1   
3        7.571845      3.965316           0                    0   
4        3.049807      8.260605           0                    0   

   HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
0                   0       0         0                       0   
1                   1       0         0                       0   
2                   1       0         1                       0   
3                   0       0         1                       0   
4                   0       0         1                       0   

   LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
0          1.369051         4.941206         0                  0   
1          2.197767         1.702393         1                  0   
2          1.698011         5.022553         1                  1   
3          3.032037         2.300159         1                  0   
4          3.470589         3.067944         1                  1   

   ChestTightness  Coughing  NighttimeSymptoms  ExerciseInduced  Diagnosis  \
0               1         0                  0                1          0   
1               0         1                  1                1          0   
2               1         0                  1                1          0   
3               1         1                  1                0          0   
4               1         0                  0                1          0   

  DoctorInCharge  
0      Dr_Confid  
1      Dr_Confid  
2      Dr_Confid  
3      Dr_Confid  
4      Dr_Confid

Buyerda endi "Age" ustun yo'q. Biz uni tashlab yubordik: (drop = True) oqali. Default holatda drop = False ga teng bo'ladi

### Agarda biz aniq biron bir qatorni ko'rmoqchi bo'lsak (isin) funksiyasidan foydalansak bo'ladi

In [27]:
df[df["Age"].isin(range(20,22))]

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
5          5039   21       0          2               0  21.812975        0   
32         5066   21       0          0               3  29.161877        0   
70         5104   20       1          0               1  26.470969        0   
72         5106   21       1          0               1  33.172248        1   
129        5163   21       1          1               1  34.055987        0   
...         ...  ...     ...        ...             ...        ...      ...   
2214       7248   20       0          0               2  35.110856        0   
2223       7257   20       0          0               2  37.790013        1   
2232       7266   21       1          0               1  33.547486        0   
2345       7379   20       0          0               1  21.056607        0   
2357       7391   20       1          1               0  19.176039        0   

      PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
5             0.470044     1.759118      9.549262           1.711446   
32            5.438137     6.220673      5.619865           7.740486   
70            0.633245     8.201199      9.235679           7.162327   
72            6.879082     1.075314      5.236114           5.169310   
129           5.834472     3.115390      4.546817           3.196260   
...                ...          ...           ...                ...   
2214          7.528743     2.328523      8.772971           7.172629   
2223          2.302761     0.570565      6.242517           5.250513   
2232          5.988057     1.280764      9.516626           4.666365   
2345          3.317995     4.155262      9.616201           5.098364   
2357          1.358787     6.207399      6.590314           1.895321   

      PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
5           7.192424      6.832048           1                    0   
32          7.651415      4.558027           0                    0   
70          5.159162      1.674201           0                    0   
72          5.972412      0.128188           1                    0   
129         5.785487      2.130553           1                    0   
...              ...           ...         ...                  ...   
2214        7.835260      8.255621           0                    0   
2223        9.760967      7.905787           0                    0   
2232        5.786901      4.955093           0                    1   
2345        5.495207      3.389376           0                    0   
2357        2.954492      6.909502           0                    0   

      HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
5                      0       1         0                       0   
32                     0       0         0                       0   
70                     0       1         0                       0   
72                     0       0         0                       0   
129                    1       0         0                       0   
...                  ...     ...       ...                     ...   
2214                   1       0         0                       0   
2223                   1       0         0                       0   
2232                   0       0         0                       0   
2345                   1       0         0                       0   
2357                   0       0         0                       0   

      LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
5             2.328191         5.898515         1                  0   
32            2.363484         5.270717         0                  0   
70            3.982546         1.719228         0                  0   
72            3.785853         5.733351         1                  1   
129           2.415917         4.164724         0                  0   
...                ...              ...       ...                .

In [28]:
df[df["Age"].isin([20,21])]

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
5          5039   21       0          2               0  21.812975        0   
32         5066   21       0          0               3  29.161877        0   
70         5104   20       1          0               1  26.470969        0   
72         5106   21       1          0               1  33.172248        1   
129        5163   21       1          1               1  34.055987        0   
...         ...  ...     ...        ...             ...        ...      ...   
2214       7248   20       0          0               2  35.110856        0   
2223       7257   20       0          0               2  37.790013        1   
2232       7266   21       1          0               1  33.547486        0   
2345       7379   20       0          0               1  21.056607        0   
2357       7391   20       1          1               0  19.176039        0   

      PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
5             0.470044     1.759118      9.549262           1.711446   
32            5.438137     6.220673      5.619865           7.740486   
70            0.633245     8.201199      9.235679           7.162327   
72            6.879082     1.075314      5.236114           5.169310   
129           5.834472     3.115390      4.546817           3.196260   
...                ...          ...           ...                ...   
2214          7.528743     2.328523      8.772971           7.172629   
2223          2.302761     0.570565      6.242517           5.250513   
2232          5.988057     1.280764      9.516626           4.666365   
2345          3.317995     4.155262      9.616201           5.098364   
2357          1.358787     6.207399      6.590314           1.895321   

      PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
5           7.192424      6.832048           1                    0   
32          7.651415      4.558027           0                    0   
70          5.159162      1.674201           0                    0   
72          5.972412      0.128188           1                    0   
129         5.785487      2.130553           1                    0   
...              ...           ...         ...                  ...   
2214        7.835260      8.255621           0                    0   
2223        9.760967      7.905787           0                    0   
2232        5.786901      4.955093           0                    1   
2345        5.495207      3.389376           0                    0   
2357        2.954492      6.909502           0                    0   

      HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
5                      0       1         0                       0   
32                     0       0         0                       0   
70                     0       1         0                       0   
72                     0       0         0                       0   
129                    1       0         0                       0   
...                  ...     ...       ...                     ...   
2214                   1       0         0                       0   
2223                   1       0         0                       0   
2232                   0       0         0                       0   
2345                   1       0         0                       0   
2357                   0       0         0                       0   

      LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
5             2.328191         5.898515         1                  0   
32            2.363484         5.270717         0                  0   
70            3.982546         1.719228         0                  0   
72            3.785853         5.733351         1                  1   
129           2.415917         4.164724         0                  0   
...                ...              ...       ...                .

ikkala jadval ham bitta narsa, faqat bittasida range(oraliq beramiz) beramiz

### Qatorlarni ko'rishni boshqa yo'li, Bu yerda index bilan ishlanadi

In [29]:
df_ID.loc[range(6024,6031)]

Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
PatientID                                                               
6024         5       1          3               1  36.251830        0   
6025        69       1          3               2  21.573741        0   
6026        26       0          0               2  17.960172        1   
6027         6       0          1               0  15.119142        0   
6028        62       1          1               2  38.599396        0   
6029        78       1          0               1  37.377390        0   
6030        31       0          0               1  38.976901        1   

           PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
PatientID                                                                   
6024               8.847617     3.270534      7.442324           0.351468   
6025               2.325817     5.123585      9.816018           0.295290   
6026               1.412605     1.822480      7.581420           8.349502   
6027               5.697078     7.350453      7.079975           5.635669   
6028               9.294254     2.627527      8.460861           2.368196   
6029               8.837949     6.346936      8.624223           4.834422   
6030               1.024265     8.292356      6.452185           9.213678   

           PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
PatientID                                                                  
6024             6.322349      8.676209           0                    1   
6025             7.834095      6.183596           1                    0   
6026             5.739466      1.182606           0                    0   
6027             2.113581      9.082656           0                    0   
6028             4.992538      8.263830           0                    0   
6029             2.328120      7.755876           0                    1   
6030             6.337221      1.796556           0                    1   

           HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
PatientID                                                                 
6024                        1       1         0                       0   
6025                        1       0         0                       0   
6026                        1       0         0                       0   
6027                        0       0         0                       0   
6028                        0       0         0                       0   
6029                        0       1         0                       0   
6030                        0       0         1                       0   

           LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
PatientID                                                                   
6024               3.020129         2.523197         1                  1   
6025               2.747082         2.666431         1                  0   
6026               2.662147         3.001780         0                  0   
6027               1.897769         2.103834         1                  0   
6028               3.324074         4.402348         1                  0   
6029               1.940728         5.663795         1                  0   
6030               3.374273         2.846202         1                  0   

           ChestTightness  Coughing  NighttimeSymptoms  ExerciseInduced  \
PatientID                                                                 
6024                    0         0                  1                1   
6025                    0         1                  1                1   
6026                    1         0                  1                0   
6027                    0         0                  1                0   
6028                    1         0                  1                1   
6029                    1         1                  1                1   
6030

Yuqorida 'PatientID' ni index qilib olgan edik. "loc" orqali unga murojaat qilyabmiz. "loc" orqali murojaat qilayotganimizda index noyob bo'lishi shart emas.

In [30]:
df_Age.loc[5].head()

PatientID  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
Age                                                                     
5         5097       1          0               1  24.689399        0   
5         5171       0          0               2  33.348863        0   
5         5439       0          3               2  21.866782        0   
5         5494       1          0               1  26.804660        0   
5         5653       1          0               2  25.455246        0   

     PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
Age                                                                   
5            9.297025     7.370863      6.803392           2.863860   
5            7.163524     1.075775      4.126640           8.848051   
5            1.125857     4.762594      6.870321           8.405203   
5            2.906699     4.290695      7.894338           8.980784   
5            7.003770     3.139850      5.745023           9.168436   

     PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
Age                                                                  
5          2.929226      0.447074           0                    0   
5          0.647391      0.851207           0                    0   
5          4.833562      5.392145           1                    1   
5          8.554417      2.191782           0                    0   
5          3.733436      0.968317           0                    0   

     HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
Age                                                                 
5                     0       0         1                       0   
5                     0       0         0                       0   
5                     0       0         0                       0   
5                     0       0         0                       0   
5                     0       0         0                       0   

     LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
Age                                                                   
5            3.752622         2.647257         1                  1   
5            2.115246         5.328038         1                  1   
5            1.004808         3.550239         0                  0   
5            2.321854         2.013079         0                  1   
5            2.325320         4.643632         0                  0   

     ChestTightness  Coughing  NighttimeSymptoms  ExerciseInduced  Diagnosis  \
Age                                                                            
5                 1         0                  0                1          0   
5                 0         1                  1                1          0   
5                 1         0                  1                1          0   
5                 1         1                  1                0          0   
5                 0         0                  0                1          0   

    DoctorInCharge  
Age                 
5        Dr_Confid  
5        Dr_Confid  
5        Dr_Confid  
5        Dr_Confid  
5        Dr_Confid

In [31]:
df.set_index("Ethnicity").loc[[2,3]]

PatientID  Age  Gender  EducationLevel        BMI  Smoking  \
Ethnicity                                                               
2               5035   26       1               2  22.757042        0   
2               5036   57       0               1  18.395396        0   
2               5037   40       1               1  38.515278        0   
2               5039   21       0               0  21.812975        0   
2               5054   40       1               1  23.588420        1   
...              ...  ...     ...             ...        ...      ...   
3               7375   13       1               1  21.091498        0   
3               7383   61       0               2  39.974018        0   
3               7405   12       1               2  29.798041        0   
3               7410   75       0               0  32.319521        0   
3               7423   54       0               2  37.079560        0   

           PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
Ethnicity                                                                   
2                  5.897329     6.341014      5.153966           1.969838   
2                  6.739367     9.196237      6.840647           1.460593   
2                  1.404503     5.826532      4.253036           0.581905   
2                  0.470044     1.759118      9.549262           1.711446   
2                  9.024920     0.911513      9.913215           2.452710   
...                     ...          ...           ...                ...   
3                  8.832518     6.057046      6.789268           9.800057   
3                  6.845534     5.546791      4.617501           6.220617   
3                  3.593522     7.909654      4.861379           9.725283   
3                  6.013515     9.652002      7.784604           8.614211   
3                  4.735169     8.214064      7.483521           2.794847   

           PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
Ethnicity                                                                  
2                7.457665      6.584631           0                    0   
2                1.448189      5.445799           0                    1   
2                7.571845      3.965316           0                    0   
2                7.192424      6.832048           1                    0   
2                7.855706      3.065639           0                    0   
...                   ...           ...         ...                  ...   
3                4.318774      4.500174           0                    1   
3                7.584865      8.896731           0                    0   
3                2.509359      4.584930           0                    0   
3                6.593218      8.962963           0                    0   
3                3.055139      9.484013           0                    0   

           HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
Ethnicity                                                                 
2                           1       0         0                       0   
2                           1       0         1                       0   
2                           0       0         1                       0   
2                           0       1         0                       0   
2                           0       0         0                       0   
...                       ...     ...       ...                     ...   
3                           1       0         0                       0   
3                           1       1         0                       1   
3                           1       1         1                       0   
3                           0       1         0                       0   
3                           0       0         1                       0   

           LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
Ethnicity        

#### Ikkita uchta ustunlar bilan ham qo'llash mumkinmi?

In [32]:
df.set_index(["Age", "Ethnicity"]).loc[[(5,3),(6,1),(8,2),(11,3)]]

PatientID  Gender  EducationLevel        BMI  Smoking  \
Age Ethnicity                                                          
5   3               5439       0               2  21.866782        0   
    3               6024       1               1  36.251830        0   
    3               6175       0               2  28.406951        0   
6   1               5508       0               3  24.329249        0   
    1               5547       0               3  36.985583        0   
    1               6006       1               2  27.359521        0   
    1               6027       0               0  15.119142        0   
    1               7268       1               2  35.506181        0   
8   2               5409       1               2  35.400080        0   
    2               5437       0               1  30.628120        0   
    2               6062       0               2  27.696773        0   
    2               6538       0               1  38.530080        0   
    2               6932       1               1  23.621074        0   
    2               7033       0               3  35.997337        0   
11  3               6044       0               0  32.500482        0   
    3               7267       1               2  37.710656        1   

               PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
Age Ethnicity                                                                   
5   3                  1.125857     4.762594      6.870321           8.405203   
    3                  8.847617     3.270534      7.442324           0.351468   
    3                  9.466542     6.110586      7.502416           4.166609   
6   1                  4.646806     3.872408      6.222042           2.618920   
    1                  2.594073     5.493144      9.016675           7.352011   
    1                  6.620172     3.073849      5.544405           9.449187   
    1                  5.697078     7.350453      7.079975           5.635669   
    1                  1.869221     2.921019      6.701817           5.286804   
8   2                  8.647831     2.793232      8.665149           8.550402   
    2                  9.954117     2.924781      9.281142           5.862351   
    2                  4.510544     0.978105      4.994442           1.267932   
    2                  9.082003     8.058458      7.839812           1.950599   
    2                  6.914589     2.069162      8.127735           2.648355   
    2                  7.600753     0.636868      5.410884           4.814441   
11  3                  0.220175     2.032527      5.988609           3.145420   
    3                  6.963149     2.971045      4.768524           0.130399   

               PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
Age Ethnicity                                                                  
5   3                4.833562      5.392145           1                    1   
    3                6.322349      8.676209           0                    1   
    3                7.123877      4.168801           0                    1   
6   1                2.581469      8.569669           1                    0   
    1                4.665342      4.790202           0                    0   
    1                4.512340      6.059066           0                    1   
    1                2.113581      9.082656           0                    0   
    1                7.611491      7.327880           1                    0   
8   2                2.949084      8.952562           0                    0   
    2                5.543099      7.404575           0                    0   
    2                6.647688      8.302376           1                    0   
    2                9.448708      5.797996           1                    0   
    2                4.340939      5.907207           0                    0   
    2                2.759470      4.849975           0    

Bu yerda birinchi "Yoshi" dan tanlab oladi keyin "Etnik kelib chiqishi" dan, Masalan birinchi 5 yoshdagilarni oladi va 3 - boshqa etnik kelib chiqishga mansub bo'lganlarni oladi. va hokazo



Etnik kelib chiqish:
0: Kavkaz
1: afro-amerikalik
2: Osiyo
3: Boshqa

### Index bo'yicha tartiblashni(sort) ko'ramiz

In [33]:
df_BMI.sort_index()

PatientID  Age  Gender  Ethnicity  EducationLevel  Smoking  \
BMI                                                                     
15.031803       5419   32       1          0               2        0   
15.035989       7054   31       0          2               1        1   
15.038094       5181   27       0          1               1        0   
15.045955       5673   55       0          0               0        0   
15.054685       5805   29       0          1               1        1   
...              ...  ...     ...        ...             ...      ...   
39.931521       5725   72       1          0               1        0   
39.939589       5077   47       0          0               2        0   
39.974018       7383   61       0          3               2        0   
39.982129       5881   39       1          2               1        0   
39.985611       6128   44       0          0               0        0   

           PhysicalActivity  DietQuality  SleepQuality  PollutionExposure  \
BMI                                                                         
15.031803          0.041564     1.865203      6.183893           4.132096   
15.035989          8.735920     6.818799      7.161621           4.603509   
15.038094          1.117471     0.749281      7.830559           6.255099   
15.045955          8.121762     4.966114      8.308117           5.938908   
15.054685          6.178646     8.825090      5.581614           7.925161   
...                     ...          ...           ...                ...   
39.931521          3.928550     1.182262      4.005979           6.543496   
39.939589          0.966696     5.369913      8.633141           7.031108   
39.974018          6.845534     5.546791      4.617501           6.220617   
39.982129          4.968360     9.589524      8.179245           5.413377   
39.985611          8.839828     7.888400      6.890556           3.311518   

           PollenExposure  DustExposure  PetAllergy  FamilyHistoryAsthma  \
BMI                                                                        
15.031803        7.412071      9.705220           1                    0   
15.035989        6.306548      4.129431           0                    0   
15.038094        3.272265      0.257197           0                    0   
15.045955        4.295812      6.980671           1                    0   
15.054685        5.362061      9.033883           0                    0   
...                   ...           ...         ...                  ...   
39.931521        4.931496      3.637206           0                    1   
39.939589        5.251895      4.876249           1                    0   
39.974018        7.584865      8.896731           0                    0   
39.982129        5.099940      7.850534           0                    1   
39.985611        2.527284      2.010431           0                    1   

           HistoryOfAllergies  Eczema  HayFever  GastroesophagealReflux  \
BMI                                                                       
15.031803                   0       0         1                       0   
15.035989                   1       0         0                       0   
15.038094                   0       0         0                       0   
15.045955                   1       0         0                       0   
15.054685                   1       0         0                       0   
...                       ...     ...       ...                     ...   
39.931521                   0       0         0                       0   
39.939589                   1       0         0                       0   
39.974018                   1       1         0                       1   
39.982129                   0       0         1                       0   
39.985611                   1       1         1                       0   

           LungFunctionFEV1  LungFunctionFVC  Wheezing  ShortnessOfBreath  \
BMI              

In [34]:
df_Age_Gender_Ethnicity_EducationLevel_sorted = df.set_index(["Age","Gender", "Ethnicity", "EducationLevel"])
df_Age_Gender_Ethnicity_EducationLevel_sorted

PatientID        BMI  Smoking  \
Age Gender Ethnicity EducationLevel                                  
63  0      1         0                    5034  15.848744        0   
26  1      2         2                    5035  22.757042        0   
57  0      2         1                    5036  18.395396        0   
40  1      2         1                    5037  38.515278        0   
61  0      0         3                    5038  19.283802        0   
...                                        ...        ...      ...   
43  1      0         2                    7421  29.059613        0   
18  1      0         1                    7422  20.740850        0   
54  0      3         2                    7423  37.079560        0   
46  1      0         2                    7424  23.444712        0   
26  1      0         0                    7425  28.123021        1   

                                     PhysicalActivity  DietQuality  \
Age Gender Ethnicity EducationLevel                                  
63  0      1         0                       0.894448     5.488696   
26  1      2         2                       5.897329     6.341014   
57  0      2         1                       6.739367     9.196237   
40  1      2         1                       1.404503     5.826532   
61  0      0         3                       4.604493     3.127048   
...                                               ...          ...   
43  1      0         2                       3.019854     6.119637   
18  1      0         1                       5.805180     4.386992   
54  0      3         2                       4.735169     8.214064   
46  1      0         2                       9.672637     7.362861   
26  1      0         0                       1.613138     7.412878   

                                     SleepQuality  PollutionExposure  \
Age Gender Ethnicity EducationLevel                                    
63  0      1         0                   8.701003           7.388481   
26  1      2         2                   5.153966           1.969838   
57  0      2         1                   6.840647           1.460593   
40  1      2         1                   4.253036           0.581905   
61  0      0         3                   9.625799           0.980875   
...                                           ...                ...   
43  1      0         2                   8.300960           2.483829   
18  1      0         1                   7.731192           7.733983   
54  0      3         2                   7.483521           2.794847   
46  1      0         2                   6.717272           9.448862   
26  1      0         0                   8.512253           3.231709   

                                     PollenExposure  DustExposure  PetAllergy  \
Age Gender Ethnicity EducationLevel                                             
63  0      1         0                     2.855578      0.974339           1   
26  1      2         2                     7.457665      6.584631           0   
57  0      2         1                     1.448189      5.445799           0   
40  1      2         1                     7.571845      3.965316           0   
61  0      0         3                     3.049807      8.260605           0   
...                                             ...           ...         ...   
43  1      0         2                     7.314582      3.425445           1   
18  1      0         1                     2.279073      6.467701           0   
54  0      3         2                     3.055139      9.484013           0   
46  1      0         2                     7.712584      5.051405           0   
26  1      0         0                     3.874028      5.064317           0   

                                     FamilyHistoryAsthma  HistoryOfAllergies  \
Age Gender Ethnicity EducationLevel                                            
63  0      1         0                                 1        

In [35]:
df_Age_Gender_Ethnicity_EducationLevel_sorted.sort_index().head(20)

PatientID        BMI  Smoking  \
Age Gender Ethnicity EducationLevel                                  
5   0      0         0                    5737  30.163020        0   
                     1                    5848  17.572297        0   
                     1                    6472  17.651456        0   
                     1                    7245  38.155312        1   
                     1                    7420  32.940790        0   
                     2                    5171  33.348863        0   
                     3                    7240  32.419103        0   
           1         0                    6860  36.172342        0   
                     1                    6613  18.601088        1   
                     1                    7382  20.936782        0   
                     2                    5756  20.942204        0   
                     2                    7258  29.857639        0   
                     3                    6323  21.819417        0   
                     3                    7115  23.508565        0   
           2         1                    5713  26.850546        0   
                     2                    6978  24.977192        0   
                     3                    6642  31.358812        0   
           3         2                    5439  21.866782        0   
                     2                    6175  28.406951        0   
    1      0         0                    6818  19.462619        1   

                                     PhysicalActivity  DietQuality  \
Age Gender Ethnicity EducationLevel                                  
5   0      0         0                       2.048262     0.123081   
                     1                       3.750060     7.641404   
                     1                       4.372488     0.794238   
                     1                       5.494052     5.390795   
                     1                       8.705633     2.110108   
                     2                       7.163524     1.075775   
                     3                       5.719808     0.661500   
           1         0                       5.417269     4.478271   
                     1                       2.182973     3.723267   
                     1                       5.277747     2.910634   
                     2                       7.725394     6.835943   
                     2                       0.225345     2.179544   
                     3                       0.759969     7.928065   
                     3                       7.574201     0.331971   
           2         1                       6.683361     3.252095   
                     2                       4.954069     9.906439   
                     3                       4.060767     2.887491   
           3         2                       1.125857     4.762594   
                     2                       9.466542     6.110586   
    1      0         0                       4.161883     8.511220   

                                     SleepQuality  PollutionExposure  \
Age Gender Ethnicity EducationLevel                                    
5   0      0         0                   4.619742           0.051558   
                     1                   8.642893           1.627050   
                     1                   4.902879           9.793511   
                     1                   7.125330           7.046773   
                     1                   9.261652           9.683211   
                     2                   4.126640           8.848051   
                     3                   5.857242           4.066618   
           1         0                   5.777935           9.409044   
                     1                   5.681424           1.947672   
                     1                   9.553447           2.197261   
                     2                   9.866307           5.707830   
                   

'Age	-> Gender -> Ethnicity	-> EducationLevel'	ketma-ketligi bo'yicha index bo'yicha sortlash

In [36]:
df_AGEE = df_Age_Gender_Ethnicity_EducationLevel_sorted
df_AGEE.sort_index(level = ["Gender", "Age", "Ethnicity"], ascending = [True, False, False]).head(20)

PatientID        BMI  Smoking  \
Age Gender Ethnicity EducationLevel                                  
79  0      3         1                    5096  19.164467        0   
                     1                    5260  18.372529        0   
                     2                    6419  36.136132        0   
           2         1                    5305  20.455018        0   
                     1                    6589  20.116128        0   
           1         1                    5877  34.770417        0   
                     2                    6555  21.304174        0   
                     2                    6645  38.712137        1   
                     0                    7210  33.791550        0   
           0         3                    5102  17.786832        0   
                     1                    5328  31.641949        0   
                     1                    5358  35.104255        0   
                     0                    6410  39.666881        0   
                     2                    6675  30.030769        0   
                     2                    6811  28.439224        1   
                     1                    6919  26.507618        0   
                     0                    6967  38.578388        1   
78  0      3         2                    6359  33.839084        0   
           2         0                    5350  38.982826        0   
                     3                    5705  25.863944        0   

                                     PhysicalActivity  DietQuality  \
Age Gender Ethnicity EducationLevel                                  
79  0      3         1                       9.586025     3.983649   
                     1                       5.427240     2.334628   
                     2                       8.778135     6.816322   
           2         1                       8.557270     5.690571   
                     1                       9.530659     7.391920   
           1         1                       5.755839     7.581654   
                     2                       6.319441     5.769623   
                     2                       5.689947     1.170514   
                     0                       1.150078     4.786714   
           0         3                       1.933638     0.403664   
                     1                       7.537056     1.828485   
                     1                       5.553994     2.723033   
                     0                       9.673421     6.641485   
                     2                       9.507565     9.315187   
                     2                       5.271687     5.828416   
                     1                       0.512970     5.538862   
                     0                       3.261688     6.362565   
78  0      3         2                       6.915997     2.465817   
           2         0                       4.827005     4.960565   
                     3                       5.239308     1.310357   

                                     SleepQuality  PollutionExposure  \
Age Gender Ethnicity EducationLevel                                    
79  0      3         1                   9.900538           9.945629   
                     1                   7.620781           5.392763   
                     2                   4.781656           5.190002   
           2         1                   5.889779           3.503976   
                     1                   8.311071           8.163724   
           1         1                   5.284070           2.894086   
                     2                   9.164750           8.025476   
                     2                   5.303048           0.977753   
                     0                   7.442274           8.327831   
           0         3                   4.935988           5.291189   
                     1                   6.735068           1.683005   
                   

In [37]:
df_AGEE.sort_index().loc[7:10]

PatientID        BMI  Smoking  \
Age Gender Ethnicity EducationLevel                                  
7   0      0         0                    6550  36.525744        0   
                     0                    6703  15.451204        0   
                     1                    6829  35.266744        0   
                     1                    6882  15.881548        0   
                     1                    6947  21.481851        0   
...                                        ...        ...      ...   
10  1      0         2                    7047  22.718430        1   
                     3                    7061  22.511747        0   
           1         0                    6004  38.755457        0   
           2         2                    5285  19.353679        0   
           3         2                    5872  21.092655        0   

                                     PhysicalActivity  DietQuality  \
Age Gender Ethnicity EducationLevel                                  
7   0      0         0                       5.824363     2.642379   
                     0                       8.441195     7.294995   
                     1                       0.861688     2.741472   
                     1                       9.552591     1.420912   
                     1                       6.573121     9.359268   
...                                               ...          ...   
10  1      0         2                       7.197237     8.974245   
                     3                       1.750145     3.921984   
           1         0                       5.193026     3.070502   
           2         2                       2.694537     0.229824   
           3         2                       9.707035     9.624419   

                                     SleepQuality  PollutionExposure  \
Age Gender Ethnicity EducationLevel                                    
7   0      0         0                   7.456644           5.073888   
                     0                   5.541541           8.864298   
                     1                   4.351592           8.983096   
                     1                   5.757846           0.763517   
                     1                   9.042883           2.176970   
...                                           ...                ...   
10  1      0         2                   4.623749           6.488494   
                     3                   9.913659           5.111092   
           1         0                   7.450705           2.919576   
           2         2                   8.419609           0.815969   
           3         2                   6.588944           3.286193   

                                     PollenExposure  DustExposure  PetAllergy  \
Age Gender Ethnicity EducationLevel                                             
7   0      0         0                     0.234759      7.939221           0   
                     0                     0.638237      8.726809           0   
                     1                     8.203493      6.560380           0   
                     1                     7.439651      7.343378           0   
                     1                     4.845059      0.700974           0   
...                                             ...           ...         ...   
10  1      0         2                     5.299605      8.829038           0   
                     3                     8.799936      2.647520           0   
           1         0                     7.197622      1.714968           0   
           2         2                     4.155306      1.007617           0   
           3         2                     6.123882      0.086528           0   

                                     FamilyHistoryAsthma  HistoryOfAllergies  \
Age Gender Ethnicity EducationLevel                                            
7   0      0         0                                 0        

"Age" bo'yicha sort qilib 7 dan 10 yoshgacha bo'lganlarni ko'rish imkoniyatiga ega bo'ldik. Sort qilmagan paytimizda xatolik bo'ldi.

In [38]:
df_AGEE.sort_index().loc[(7,2):(10,1)]

PatientID        BMI  Smoking  \
Age Gender Ethnicity EducationLevel                                  
8   0      0         0                    5147  27.207777        0   
                     0                    6297  20.033355        0   
                     0                    6504  20.021081        0   
                     1                    6414  27.951586        0   
                     1                    6652  24.268124        0   
...                                        ...        ...      ...   
10  1      0         2                    7047  22.718430        1   
                     3                    7061  22.511747        0   
           1         0                    6004  38.755457        0   
           2         2                    5285  19.353679        0   
           3         2                    5872  21.092655        0   

                                     PhysicalActivity  DietQuality  \
Age Gender Ethnicity EducationLevel                                  
8   0      0         0                       5.580948     8.042787   
                     0                       0.031091     8.982038   
                     0                       3.663363     2.634147   
                     1                       0.481628     5.174009   
                     1                       9.496826     8.811092   
...                                               ...          ...   
10  1      0         2                       7.197237     8.974245   
                     3                       1.750145     3.921984   
           1         0                       5.193026     3.070502   
           2         2                       2.694537     0.229824   
           3         2                       9.707035     9.624419   

                                     SleepQuality  PollutionExposure  \
Age Gender Ethnicity EducationLevel                                    
8   0      0         0                   9.041389           8.976515   
                     0                   5.719565           9.352514   
                     0                   8.985468           4.144855   
                     1                   9.392533           1.750450   
                     1                   7.962185           2.808034   
...                                           ...                ...   
10  1      0         2                   4.623749           6.488494   
                     3                   9.913659           5.111092   
           1         0                   7.450705           2.919576   
           2         2                   8.419609           0.815969   
           3         2                   6.588944           3.286193   

                                     PollenExposure  DustExposure  PetAllergy  \
Age Gender Ethnicity EducationLevel                                             
8   0      0         0                     6.700115      2.347808           0   
                     0                     9.828016      6.417503           0   
                     0                     6.967410      8.453155           0   
                     1                     9.053773      9.516464           0   
                     1                     2.819765      6.505102           0   
...                                             ...           ...         ...   
10  1      0         2                     5.299605      8.829038           0   
                     3                     8.799936      2.647520           0   
           1         0                     7.197622      1.714968           0   
           2         2                     4.155306      1.007617           0   
           3         2                     6.123882      0.086528           0   

                                     FamilyHistoryAsthma  HistoryOfAllergies  \
Age Gender Ethnicity EducationLevel                                            
8   0      0         0                                 0        

"Age" va "Gender" bo'yicha sort qilib 7 dan(ayollar) 10 (erkaklar) yoshgacha bo'lganlarni ko'rish imkoniyatiga ega bo'ldik. Sort qilmagan paytimizda xatolik bo'ldi.

In [39]:
df_AGEE.sort_index().loc[:, "PatientID":"Smoking"]

PatientID        BMI  Smoking
Age Gender Ethnicity EducationLevel                               
5   0      0         0                    5737  30.163020        0
                     1                    5848  17.572297        0
                     1                    6472  17.651456        0
                     1                    7245  38.155312        1
                     1                    7420  32.940790        0
...                                        ...        ...      ...
79  1      1         2                    5105  38.421666        0
                     2                    5996  24.625133        1
           3         0                    7018  17.867079        0
                     1                    6429  30.140627        0
                     2                    5818  15.325347        0

[2392 rows x 3 columns]

indexlarni o'zgarmasdan qoldirib, to'gri ustunlarni filtr qildik

### iloc bilan ishlash

In [40]:
df.iloc[2:19, 1:6]

Age  Gender  Ethnicity  EducationLevel        BMI
2    57       0          2               1  18.395396
3    40       1          2               1  38.515278
4    61       0          0               3  19.283802
5    21       0          2               0  21.812975
6    45       1          1               1  30.245954
7    26       0          0               1  26.048416
8    49       1          1               2  32.676204
9    45       1          1               1  29.910298
10   27       0          1               2  32.624314
11   62       1          0               1  35.288859
12   61       1          0               0  20.347040
13   22       0          0               2  34.181508
14   43       0          0               2  35.045684
15   46       1          1               2  19.198629
16    7       1          0               1  31.695390
17   27       1          1               1  35.547056
18   48       1          0               2  28.440465

'iloc' ni biz indexlarni standart default xolati bilan ishlayotganda ishlashda qulay